app/controllers/checkouts_controller.rb

```diff
@@ -3,13 +3,26 @@ class CheckoutsController < ApplicationController
 
   def create
     @book = Book.find(params[:book_id])
-    checkout = Checkout.create(book: @book, user: current_user)
+    if CheckOutBook.process(book: @book, user: current_user)
+      flash[:success] = t('.successful_message')
+      redirect_to root_path
+    else
+      flash[:error] = t('.unsuccessful_message')
```

Comment:
I understand now. Because I did not specify a redirect in the else clause, it will try to render create, which will fail. A failure I will never catch because I don't know how test the scenario where someone else checked the book out right before you did. I removed both `else` clauses and added `create!` and `update!` in `RenewBook` and `CheckOutBook` services respectively instead.